graph/appender/dead_node.go

```diff
@@ -21,16 +27,54 @@ func (a DeadNodeAppender) AppendGraph(trafficMap graph.TrafficMap, _ string) {
 	business, err := business.Get()
 	checkError(err)
 
-	applyDeadNodes(trafficMap, business.Workload)
+	if a.externalServices == nil {
+		a.externalServices = resolveExternalServices(business)
+	}
+
+	applyDeadNodes(trafficMap, business.Workload, a.externalServices)
 }
 
-func applyDeadNodes(trafficMap graph.TrafficMap, wkSvc business.WorkloadService) {
+// resolveExternalServices queries the cluster API to resolve egress services
+// by using ServiceEntries resources across all namespaces in the cluster.
+// All ServiceEntries are needed, because Istio does not distinguish where the
+// ServiceEntries are created when routing egress traffic.
+func resolveExternalServices(bLayer *business.Layer) map[string]bool {
```

Comment:
We maybe need a way to cache this sort of information to be used repeatedly, and not redo it each time we generate a graph.  This is not a blocker, I'm just saying we need to maybe think about how to improve the handling for this slow-changing data.